pkg/capi/deploy.go

```diff
@@ -256,3 +260,36 @@ func (clusterAPI *Manager) DeployCluster(ctx context.Context, clusterName string
 
 	return deployedCluster, nil
 }
+
+// DestroyCluster deletes cluster.
+func (clusterAPI *Manager) DestroyCluster(ctx context.Context, name, namespace, version string) error {
+	cluster := &unstructured.Unstructured{}
+	cluster.SetName(name)
+	cluster.SetNamespace(namespace)
+	cluster.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "cluster.x-k8s.io",
+		Kind:    "Cluster",
+		Version: version,
+	})
+
+	if err := clusterAPI.runtimeClient.Delete(ctx, cluster); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	return retry.Constant(30*time.Minute, retry.WithUnits(10*time.Second), retry.WithErrorLogging(true)).Retry(func() error {
+		err := clusterAPI.runtimeClient.Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, cluster)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil
+			}
+
+			return retry.ExpectedError(err)
```

Comment:
I would argue it's `UnexpectedError` here, or just `return err`

Reply:
Well... Maybe, was too fast to merge it. Sorry. Will change it in some subsequent PR.